beacon-chain/db/attestation.go

```diff
@@ -43,6 +54,28 @@ func (db *BeaconDB) Attestation(hash [32]byte) (*pb.Attestation, error) {
 	return attestation, err
 }
 
+// Attestation retrieves all the attestation records from the db.
+func (db *BeaconDB) Attestations() ([]*pb.Attestation, error) {
+	var attestations []*pb.Attestation
+	err := db.view(func(tx *bolt.Tx) error {
+		a := tx.Bucket(attestationBucket)
+
+		if err := a.ForEach(func(k, v []byte) error {
+			attestation, err := createAttestation(v)
+			if err != nil {
+				return err
+			}
+			attestations = append(attestations, attestation)
```

Comment:
Why don't we cap it off at max attestations here? What if we have a million records in the db, this method will fetch them all

Reply:
I'd rather keep this function as generic as possible so anyone can use it w/o having to worry it's capped or not. I think the clear separation of responsibility is nice (DB retrieves it, and Operation service caps it).
In terms of millions records in the DB, we have clean up routines that deletes attestation after it's seen from the block, and we only keep attestations up to last finalized block.

Reply:
Will update comment to be my explicit, these are the attestations that have not been included on chain

Reply:
Fair enough, I see the use case where we might want to fetch all of them perhaps for visualizations or metrics gathering